examples/error-destructure-binding-pattern/error_destructure_binding_pattern.bal

```diff
@@ -0,0 +1,45 @@
+import ballerina/io;
+
+type SampleError error<string, map<anydata|error>>;
+
+public function main() {
+    string reason;
+    map<anydata|error> detail;
+    // This error de-structure binding pattern will de-structure an `error` value of type `SampleError`
```

Comment:
Shall we say "destructure" instead of "de-structure" since that is what is used in the spec?

```suggestion
    // This error destructure binding pattern will destructure an `error` value of type `SampleError`
```